MVC/models/articlesModels.js

```diff
@@ -11,53 +11,70 @@ exports.selectArticleById = (article_id) => {
   });
 };
 
-exports.selectAllArticles = () => {
-  const sql = `SELECT articles.article_id, articles.title, articles.topic, 
+exports.selectAllArticles = (topic) => {
+  const arrTopics = ["mitch", "cats", "paper"];
+
+  let sql = `SELECT articles.article_id, articles.title, articles.topic, 
     articles.author, articles.created_at, articles.votes, articles.article_img_url, 
     COUNT(comments.comment_id) AS comment_count FROM articles
     LEFT JOIN comments ON articles.article_id = comments.article_id
     GROUP BY articles.article_id
     ORDER BY articles.created_at DESC;`;
+
+  if (topic && !arrTopics.includes(topic)) {
+    return Promise.reject({ status: 404, msg: "Topic not found" });
+  } else if (topic && arrTopics.includes(topic)) {
+    sql = `SELECT articles.article_id, articles.title, articles.topic, 
```

Comment:
This isn't a very DRY solution, it would be better to conditionally build a string based on some conditional logic rather than rewriting the whole query.